### PR TITLE
Fixes to live combine trigger fits

### PR DESCRIPTION
**Pull request by GarethDaviesGW**
Allow date format to not be YYYY_MM_DD, but (e.g.) YYYY_M_D
fix issue which would break things if no triggers are found in the duration bin

Note that this does not affect the validity of reviews - it is mostly due to dealing with the O2 triggers

### CODE REVIEW

**Comment by GarethDaviesGW** [comment]
(And no rush on this - obviously paper takes precedence)

File: bin/live/pycbc_live_combine_single_fits
Comment (titodalcanton) [inline bin/live/pycbc_live_combine_single_fits:79]: I am a bit confused by the parentheses in the RE and by the split on underscores. Can you not just do this:
```python
re_string = '([12]\d{3})_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01])'
m = re.search(re_string, f)
analysis_year = int(m.group(1))
analysis_month = int(m.group(2))
analysis_day = int(m.group(3))
```

**Comment by titodalcanton** [comment]
I cannot speak for the functionality (I assume you tested it), but modulo the comment above this looks good.

File: bin/live/pycbc_live_combine_single_fits
Comment (GarethDaviesGW) [reply bin/live/pycbc_live_combine_single_fits:79]: that is something I didn't know you could do. That makes more sense

Comment (titodalcanton) [reply bin/live/pycbc_live_combine_single_fits]: Makes more sense now, but if my parsing of the RE is correct, should it not be
```
([12]\d{3})_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01])
```
instead of
```
([12]\d{3}_(0?[1-9]|1[0-2])_(0?[1-9]|[12]\d|3[01]))
```
(a subtle difference in parentheses)?

**Comment by ahnitz** [comment]
@GarethDaviesGW @titodalcanton What's the status of this?

**Comment by titodalcanton** [comment]
I think I still have an open question to @GarethDaviesGW about the correctness of the regexp.

**Comment by GarethDaviesGW** [comment]
That's right - I had just forgotten about this. I'll implement Tito's
suggestion and then merge

**Comment by GarethDaviesGW** [comment]
(well I'll merge after approval)

**Comment by GarethDaviesGW** [comment]
oops - that was the wrong button

**Review by titodalcanton — APPROVED** [review]
The change looks good now, but I am mystified by the Travis failure...

**Pull request merged**
